lib/pavilion/scriptcomposer.py

```diff
@@ -0,0 +1,697 @@
+#  available from LANL.
+#
+#  Additionally, redistribution and use in source and binary 
+#  forms, with or without modification, are permitted provided 
+#  that the following conditions are met:
+#  -  Redistributions of source code must retain the 
+#     above copyright notice, this list of conditions 
+#     and the following disclaimer. 
+#  -  Redistributions in binary form must reproduce the 
+#     above copyright notice, this list of conditions 
+#     and the following disclaimer in the documentation 
+#     and/or other materials provided with the distribution. 
+#  -  Neither the name of Los Alamos National Security, LLC, 
+#     Los Alamos National Laboratory, LANL, the U.S. Government, 
+#     nor the names of its contributors may be used to endorse 
+#     or promote products derived from this software without 
+#     specific prior written permission.
+#   
+#  THIS SOFTWARE IS PROVIDED BY LOS ALAMOS NATIONAL SECURITY, LLC 
+#  AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+#  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+#  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. 
+#  IN NO EVENT SHALL LOS ALAMOS NATIONAL SECURITY, LLC OR CONTRIBUTORS 
+#  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, 
+#  OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, 
+#  OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY 
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR 
+#  TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT 
+#  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY 
+#  OF SUCH DAMAGE.
+#
+#  ###################################################################
+
+from __future__ import unicode_literals
+import os, datetime, grp, pwd, stat
+from collections import OrderedDict
+
+""" Class to allow for scripts to be written for other modules.
+    Typically, this will be used to write bash or batch scripts. 
+"""
+
+
+class scriptHeader( object ):
+    """Class to serve as a struct for the script header."""
+
+    @property
+    def shell_path( self ):
+        """Function to return the value of the internal shell path variable."""
+        return self._shell_path
+
+    @shell_path.setter
+    def shell_path( self, value ):
+        """Function to set the value of the internal shell path variable."""
+        if value != None and not isinstance(value, unicode):
+            error = "Shell Path must be of type 'str', not {}".format(
+                    type( value ) )
+            raise TypeError( error )
+
+        self._shell_path = value
+
+    @property
+    def scheduler_macros( self ):
+        """Function to return the value of the internal scheduler macros
+        variable.
+        """
+        return self._scheduler_macros
+
+    @scheduler_macros.setter
+    def scheduler_macros( self, value ):
```

Comment:
"Macro" is the wrong word here. A macro is basically a function. I'm not even sure what this is supposed to be.